lecture_1/task/main.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+type jobApplicationResult struct {
+	Name   string
+	Age    int
+	Passed bool
+	Skills []string
+}
+
+const url = "https://run.mocky.io/v3/f7ceece5-47ee-4955-b974-438982267dc8"
+
+func getPassedApplications(list []jobApplicationResult) map[string]struct{} {
+	f, err := os.Create("passed_Go_Java.txt")
+	if err != nil {
+		log.Fatal(
+			errors.WithMessage(err, "opening a file"),
+		)
+	}
+
+	set := map[string]struct{}{}
+
+	for _, applicant := range list {
+		if applicant.Passed {
+			for _, skill := range applicant.Skills {
+				if skill == "Java" || skill == "Go" {
+					set[applicant.Name] = struct{}{}
+					f.WriteString(fmt.Sprint(applicant.Name) + "\n")
+					break
+				}
+			}
+		}
+	}
```

Comment:
This is a little difficult to read. You should write your own `contains` method to check whether an applicant has a certain skill.